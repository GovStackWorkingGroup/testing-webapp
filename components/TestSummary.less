@import '../styles/Variables.less';

.test-summary-component {
  background-color: #fafbff;
  border: 1px solid @stroke;
  border-radius: 8px;
  padding: 24px;
}

.test-summary-row {
  display: grid;
  grid-template-columns: 2fr 1fr 1fr 2fr;
  grid-template-rows: 24px;
  align-items: center;
  gap: 5px;
}

.tooltipTriggerArea {
  position: absolute;
  top: 25%;
  left: 25%;
  width: 50%;
  height: 50%;
  z-index: 2; /* Ensure it's above the BBImage */
  pointer-events: none; /* Ensures that interactions pass through this div to underlying content */
}

.test-summary-header {
  color: @text-secondary;
  padding-bottom: 10px;
}

.test-summary-label {
  font-size: 16px;
}

.test-summary-bb {
  display: flex;
  flex-direction: row;
  gap: 8px;
  max-width: 90%;
}

.test-summary-bb p {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.test-summary-icon {
  filter: invert(58%) sepia(97%) saturate(8%) hue-rotate(44deg) brightness(101%)
    contrast(87%);
  padding: 0;
}
