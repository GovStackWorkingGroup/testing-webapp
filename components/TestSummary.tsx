import { useRouter } from 'next/router';
import React, { useCallback } from 'react';
import { useIntl } from 'react-intl';
import BBImage from './table/BuildingBlocksImage';

const TestSummary = () => {
  const router = useRouter();
  const { formatMessage } = useIntl();

  const format = useCallback(
    (id: string) => formatMessage({ id }),
    [formatMessage]
  );

  const { buildingBlock, bbId } = router.query;

  return (
    <div className='test-summary-component' data-testid='test-summary-component'>
      <div className='test-summary-row test-summary-header'>
        <div data-testid='test-summary-bb-header'>{`${format('building_block.label')}:`}</div>
        <div data-testid='test-summary-passed-header'>{`${format('app.tests_passed.label')}:`}</div>
        <div data-testid='test-summary-failed-header'>{`${format('app.tests_failed.label')}:`}</div>
        <div data-testid='test-summary-compatibility-header'>{`${format('app.compatibility.label')}:`}</div>
      </div>
      <div className='test-summary-row test-summary-label'>
        <div className='test-summary-bb' data-testid='test-summary-bb-label'>
          <BBImage
            // @ts-ignore
            imagePath={buildingBlock}
            customStyle='test-summary-icon'
          />
          <p>{buildingBlock}</p>
        </div>
        {/* <div data-testid='test-summary-passed-label'>{testsPassed}</div>
        <div data-testid='test-summary-failed-label'>{testsFailed}</div> */}
        {/* <div data-testid='test-summary-compatibility-label'>{`${Math.floor(compatibility * 100)}%`}</div> */}
      </div>
    </div>
  );
};

export default TestSummary;
