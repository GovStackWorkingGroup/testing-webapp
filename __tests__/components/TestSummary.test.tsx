import { render } from '../test-utils/test-utils';
import TestSummary from '../../components/TestSummary';

describe('Unit tests for TestSummary component:', () => {
  const BB_HEADER_TEST_ID = 'test-summary-bb-header';
  const PASSED_TEST_HEADER_TEST_ID = 'test-summary-passed-header';
  const FAILED_TEST_HEADER_TEST_ID = 'test-summary-failed-header';
  const COMPATIBILITY_HEADER_TEST_ID = 'test-summary-compatibility-header';
  const BB_LABEL_TEST_ID = 'test-summary-bb-label';
  const PASSED_TEST_LABEL_TEST_ID = 'test-summary-passed-label';
  const FAILED_TEST_LABEL_TEST_ID = 'test-summary-failed-label';
  const COMPATIBILITY_LABEL_TEST_ID = 'test-summary-compatibility-label';

  it('should render the component and match snapshot', () => {
    const { container } = render(<TestSummary />);

    expect(container).toMatchSnapshot();
  });

  it('should have corresponding values', () => {
    const { getByTestId } = render(<TestSummary />);

    expect(getByTestId(BB_HEADER_TEST_ID)).toHaveTextContent('Building Block');
    expect(getByTestId(PASSED_TEST_HEADER_TEST_ID)).toHaveTextContent('Tests Passed');
    expect(getByTestId(FAILED_TEST_HEADER_TEST_ID)).toHaveTextContent('Tests Failed');
    expect(getByTestId(COMPATIBILITY_HEADER_TEST_ID)).toHaveTextContent('Compatibility');
    // expect(getByTestId(BB_LABEL_TEST_ID)).toHaveTextContent('mobility_management');
    // expect(getByTestId(PASSED_TEST_LABEL_TEST_ID)).toHaveTextContent('2');
    // expect(getByTestId(FAILED_TEST_LABEL_TEST_ID)).toHaveTextContent('2');
    // expect(getByTestId(COMPATIBILITY_LABEL_TEST_ID)).toHaveTextContent('34%');
  });
});
