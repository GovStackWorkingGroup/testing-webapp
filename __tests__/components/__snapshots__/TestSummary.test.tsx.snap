// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for TestSummary component: should render the component when BB is defined and match snapshot 1`] = `
<div>
  <div
    class="test-summary-component"
    data-testid="test-summary-component"
  >
    <div
      class="test-summary-row test-summary-header"
    >
      <div
        data-testid="test-summary-bb-header"
      >
        Building Block:
      </div>
      <div
        data-testid="test-summary-passed-header"
      >
        Tests Passed:
      </div>
      <div
        data-testid="test-summary-failed-header"
      >
        Tests Failed:
      </div>
      <div
        data-testid="test-summary-compatibility-header"
      >
        Compatibility:
      </div>
    </div>
    <div
      class="test-summary-row test-summary-label"
    >
      <div
        class="test-summary-bb"
        data-testid="test-summary-bb-label"
      >
        <img
          alt="Logo for: bb-payments"
          class="bb-image test-summary-icon"
          data-nimg="1"
          decoding="async"
          height="18"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2FbuildingBlocks%2Fbb-payments.png&w=48&q=75"
          srcset="/_next/image?url=%2Fimages%2FbuildingBlocks%2Fbb-payments.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2FbuildingBlocks%2Fbb-payments.png&w=48&q=75 2x"
          style="color: transparent;"
          width="18"
        />
        <p>
          Payments
        </p>
      </div>
      <div
        data-testid="test-summary-passed-label"
      >
        <p>
          2
        </p>
      </div>
      <div
        data-testid="test-summary-failed-label"
      >
        <p>
          2
        </p>
      </div>
      <div
        data-testid="test-summary-compatibility-label"
      >
        <p>
          34%
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for TestSummary component: should render the component when BB is undefined and match snapshot 1`] = `
<div>
  <div
    class="test-summary-component"
    data-testid="test-summary-component"
  >
    <div
      class="test-summary-row test-summary-header"
    >
      <div
        data-testid="test-summary-bb-header"
      >
        Building Block:
      </div>
      <div
        data-testid="test-summary-passed-header"
      >
        Tests Passed:
      </div>
      <div
        data-testid="test-summary-failed-header"
      >
        Tests Failed:
      </div>
      <div
        data-testid="test-summary-compatibility-header"
      >
        Compatibility:
      </div>
    </div>
    <div
      class="test-summary-row test-summary-label"
    >
      <div
        class="test-summary-bb"
        data-testid="test-summary-bb-label"
      >
        <div
          class="loading-bar"
        />
      </div>
      <div
        data-testid="test-summary-passed-label"
      >
        <div
          class="loading-bar"
        />
      </div>
      <div
        data-testid="test-summary-failed-label"
      >
        <div
          class="loading-bar"
        />
      </div>
      <div
        data-testid="test-summary-compatibility-label"
      >
        <div
          class="loading-bar"
        />
      </div>
    </div>
  </div>
</div>
`;
